runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsPlugin.java

```diff
@@ -121,4 +188,86 @@ private void simulateSkillChange()
 			}
 		}
 	}
+
+	private ArrayList<Widget> getMemberSkills()
+	{
+		ArrayList<Widget> membersOnlyWidgets = new ArrayList<>();
+
+		Widget agility = client.getWidget(WidgetInfo.SKILL_AGILITY),
+			construction = client.getWidget(WidgetInfo.SKILL_CONSTRUCTION),
+			farming = client.getWidget(WidgetInfo.SKILL_FARMING),
+			fletching = client.getWidget(WidgetInfo.SKILL_FLETCHING),
+			herblore = client.getWidget(WidgetInfo.SKILL_HERBLORE),
+			hunter = client.getWidget(WidgetInfo.SKILL_HUNTER),
+			thieving = client.getWidget(WidgetInfo.SKILL_THIEVING),
+			slayer = client.getWidget(WidgetInfo.SKILL_SLAYER);
+
+		membersOnlyWidgets.add(agility);
+		membersOnlyWidgets.add(construction);
+		membersOnlyWidgets.add(farming);
+		membersOnlyWidgets.add(fletching);
+		membersOnlyWidgets.add(herblore);
+		membersOnlyWidgets.add(hunter);
+		membersOnlyWidgets.add(thieving);
+		membersOnlyWidgets.add(slayer);
+
+		return membersOnlyWidgets;
+	}
+
+	private void shuffleSkills(boolean enabled)
+	{
+		if (client.getWidget(WidgetInfo.SKILL_TOTAL) == null)
+		{
+			return;
+		}
+		//move some free-to-play skills into member skill locations to fill the gaps
+		if (enabled)
+		{
+			client.getWidget(WidgetInfo.SKILL_TOTAL).setRelativeX(WOODCUTTING_X);
```

Comment:
Using some kind of class or data structure would help reduce the amount of duplicated code in this method.